hydra-node/src/Hydra/Logging/Monitoring.hs

```diff
@@ -96,16 +92,6 @@ monitor transactionsMap metricsMap = \case
     -- transactions after some timeout expires
     atomically $ modifyTVar' transactionsMap (Map.insert (txId tx) t)
     tick "hydra_head_requested_tx"
-  (Node (BeginEffect _ _ _ (ClientEffect (SnapshotConfirmed _ snapshot _)))) -> do
-    t <- getMonotonicTime
-    forM_ (confirmed snapshot) $ \tx -> do
-      txsStartTime <- readTVarIO transactionsMap
-      case Map.lookup (txId tx) txsStartTime of
-        Just start -> do
-          atomically $ modifyTVar' transactionsMap $ Map.delete (txId tx)
-          histo "hydra_head_tx_confirmation_time_ms" (diffTime t start)
-        Nothing -> pure ()
-    tickN "hydra_head_confirmed_tx" (length $ confirmed snapshot)
```

Comment:
Must: not break prometheus metrics

This is dropping metrics from the API and not scope of this PR? Or do you disagree?

Reply:
We don't have `ClientEffect` anymore. This is the only reason why this branch was removed.

Reply:
We could pattern match on `LogicOutcome` here.